### PR TITLE
Docs: Update CloudWatch and Stackdriver docs for 6.6

### PR DESCRIPTION
**Pull request by sunker**
CloudWatch: Paragraph that describes how auto period calculations are done

Stackdriver:  Paragraph the describes how to use meta data labels

### CODE REVIEW

File: docs/sources/features/datasources/stackdriver.md [hunk 1]
```diff
@@ -131,6 +131,10 @@ It is also possible to choose fixed time intervals to group by, like `1h` or `1d
 
 Group by resource or metric labels to reduce the number of time series and to aggregate the results by a group by. E.g. Group by instance_name to see an aggregated metric for a Compute instance.
 
+#### Meta data labels
+
+Meta data labels are only returned in the time series response if they're part of the group by segment in the time series request. There's no API for retrieving meta data labels, and therefore it's not possible to populate the group by dropdown with the meta data labels that are avaiable for the selected service and metric. However, the group by field dropdown comes with a pre-defined list of common system labels. User labels cannot be pre-defined, but it's possible to enter them manually in the group by field. If a meta data label, user label or system label, is included in the group by segment, it will be possible to create filters based on it and to expand its value on the alias field.
```
Comment (daniellee) [inline docs/sources/features/datasources/stackdriver.md]: I think the term `group by segment` might be hard to parse for most people. I wouldn't use the word segment. Generally this paragraph is very technical and almost assumes that people know about the underlying API. Maybe @oddlittlebird can help?

**Review by daniellee — CHANGES_REQUESTED** [review]
The period section is good but I think the Meta data labels section is too dense and hard to understand.

File: docs/sources/features/datasources/cloudwatch.md [hunk 2]
```diff
@@ -150,7 +150,7 @@ Please note that in the case you use the expression field to reference another q
 
 A period is the length of time associated with a specific Amazon CloudWatch statistic. Periods are defined in numbers of seconds, and valid values for period are 1, 5, 10, 30, or any multiple of 60.
 
-If the period field is left blank or set to `auto`, it will be calculated automatically based on the time range. The formula that is used is `time range in seconds / 2000`, and then we snap to next higher value in an array of pre-defined periods `[60, 300, 900, 3600, 21600, 86400]`. By clicking `Show Query Preview` in the query editor, you should be able to see what period was used by Grafana.
+If the period field is left blank or set to `auto`, it will be calculated automatically based on the time range. The formula that is used is `time range in seconds / 2000`, and then we snap to next higher value in an array of pre-defined periods `[60, 300, 900, 3600, 21600, 86400]`. By clicking `Show Query Preview` in the query editor, you will be able to see what period was used by Grafana.
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/cloudwatch.md]:
```suggestion
If the period field is left blank or set to `auto`, then it calculates automatically based on the time range. The formula used is `time range in seconds / 2000`, and then it snaps to the next higher value in an array of predefined periods `[60, 300, 900, 3600, 21600, 86400]`. By clicking `Show Query Preview` in the query editor, you can see what period Grafana used.
```

File: docs/sources/features/datasources/stackdriver.md (hunk 1 above)
Comment (oddlittlebird) [reply docs/sources/features/datasources/stackdriver.md]: "meta data" is one word: metadata

File: docs/sources/features/datasources/stackdriver.md [hunk 3]
```diff
@@ -133,7 +133,7 @@ Group by resource or metric labels to reduce the number of time series and to ag
...
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/stackdriver.md]:
```suggestion
Metadata labels are only returned in the time series response if they're part of the group by segment in the time series request. There's no API for retrieving metadata labels, so it's not possible to populate the group by dropdown with the metadata labels that are available for the selected service and metric. However, the group by field dropdown comes with a pre-defined list of common system labels. 

User labels cannot be pre-defined, but it's possible to enter them manually in the group by field. If a metadata label, user label or system label is included in the group by segment, then you can create filters based on it and expand its value on the alias field.
```

Comment (oddlittlebird) [reply docs/sources/features/datasources/stackdriver.md]: A few more things:

It looks like you might be talking about field names in some cases. If so, capitalize them to match what is on the screen and make them bold, per the documentation style guide.

Most of this paragraph seems to assume the reader understands the API and the metadata labels, which I unfortunately do not, so it is a bit dense for me to clearly understand what you are trying to say here.

**Review by oddlittlebird — CHANGES_REQUESTED** [review]

**Comment by sunker** [comment]
Thanks for feedback @daniellee and @oddlittlebird! I've commited the changes that you suggested Diana

I totally agree the part about metadata is dense, but I kind of feel the need to explain this since it's not as straightforward as the other labels. Some users might be wondering why there are hard coded labels in the group by dropdown, and in that case I want them to find the answer in the DS docs so that they don't have to create an issue to get the answer.

File: docs/sources/features/datasources/stackdriver.md [hunk 4]
```diff
@@ -131,6 +131,12 @@ It is also possible to choose fixed time intervals to group by, like `1h` or `1d
 
 Group by resource or metric labels to reduce the number of time series and to aggregate the results by a group by. E.g. Group by instance_name to see an aggregated metric for a Compute instance.
 
+#### Metadata labels
```
Comment (oddlittlebird) [inline docs/sources/features/datasources/stackdriver.md]: This section is much easier to read now!

Comment (sunker) [reply docs/sources/features/datasources/stackdriver.md]: Nice! Do you want to approve so that I can merge?

**Review by oddlittlebird — APPROVED** [review]

**Review by daniellee — APPROVED** [review]

**Comment by joshk0** [comment]
Sorry, I've been in a `git annotate` hole and just found this change.

I'm curious to ask about this excerpt, which survives to this day, but was written here:

> There’s no API for retrieving metadata labels. As a result, you cannot populate the group by list with the metadata labels that are available for the selected service and metric. However, the Group By field list comes with a pre-defined set of common system labels.

Do you know if this is still true today? I can see this list of metadata labels in the Metrics Explorer, but am desperately in search of at least a semi-official API method of doing that.

Thank you very much!

**Comment by sunker** [comment]
> Do you know if this is still true today? I can see this list of metadata labels in the Metrics Explorer, but am desperately in search of at least a semi-official API method of doing that.

The feedback that we got from google when implementing this two years back was that there was no api for this, but that might have changed. See the original discussion [here](https://github.com/grafana/grafana/issues/18887#issuecomment-527906987). Let us know if this has changed and we'll create an issue for removing the hardcoded values.
